add new options to `plugin-transform-react-jsx` docs (#2201)
babel/website: docs/plugin-transform-react-jsx.md

## Example

### React Automatic Runtime

Automatic runtime is a feature avaliable in v7.9.0. With this runtime enabled, the functions that JSX compiles to will be imported automatically.

**In**

```javascript
const profile = (
  <div>
    <img src="avatar.png" className="profile" />
    <h3>{[user.firstName, user.lastName].join(" ")}</h3>
  </div>
);
```

**Out**

```javascript
import { jsx as _jsx } from "react";
import { jsxs as _jsxs } from "react";

const profile = _jsxs("div", {
  children: [
    _jsx("img", {
      src: "avatar.png",
      className: "profile",
    }),
    _jsx("h3", {
      children: [user.firstName, user.lastName].join(" "),
    }),
  ],
});
```

> #### Custom for Automatic Runtime

Reviewer: Howabout:

```suggestion
#### Customizing the Runtime Import
```

Author: Oooh this sounds a lot better

**In**

```javascript
/** @jsxImportSource custom-jsx-library */

const profile = (
  <div>
    <img src="avatar.png" className="profile" />
    <h3>{[user.firstName, user.lastName].join(" ")}</h3>
  </div>
);
```

**Out**

```javascript
import { jsx as _jsx } from "custom-jsx-library";
import { jsxs as _jsxs } from "custom-jsx-library";

const profile = _jsxs("div", {
  children: [
    _jsx("img", {
      src: "avatar.png",
      className: "profile",
    }),
    _jsx("h3", {
      children: [user.firstName, user.lastName].join(" "),
    }),
  ],
});
```

**In**

```javascript
/** @jsxRuntime classic */

const profile = (
  <div>
    <img src="avatar.png" className="profile" />
    <h3>{[user.firstName, user.lastName].join(" ")}</h3>
  </div>
);
```

**Out**

```javascript
const profile = React.createElement(
  "div",
  null,
  React.createElement("img", { src: "avatar.png", className: "profile" }),
  React.createElement("h3", null, [user.firstName, user.lastName].join(" "))
);
```

### React Classic Runtime

> If you can't use auto importing, you can default to classic runtime, which is the default behavior for v7 and prior.

Reviewer: I think dropping the first "default" reads better, thoughts?

```suggestion
If you do not want (or cannot use) auto importing, you can use the classic runtime, which is the default behavior for v7 and prior.
```

**In**

```javascript
const profile = (
  <div>
    <img src="avatar.png" className="profile" />
    <h3>{[user.firstName, user.lastName].join(" ")}</h3>
  </div>
);
```

**Out**

```javascript
const profile = React.createElement(
  "div",
  null,
  React.createElement("img", { src: "avatar.png", className: "profile" }),
  React.createElement("h3", null, [user.firstName, user.lastName].join(" "))
);
```

> ### Custom for Classic Runtime

Reviewer: If we like the suggestion above, this can also match it:

```suggestion
### #### Customizing the Classic Runtime Import
```

```json
{
  "plugins": [
    [
      "@babel/plugin-transform-react-jsx",
      {
        "throwIfNamespace": false, // defaults to true
        "runtime": "automatic", // defaults to classic

>         "importSource": "Preact" // defaults to react

Reviewer 2: I think that we should avoid using this example until Preact supports it.

```suggestion
        "importSource": "custom-jsx-library" // defaults to react
```
